src/client/workflow/workflow.ts

```diff
@@ -0,0 +1,262 @@
+import { nanoid } from "nanoid";
+import type { AsyncStepFunction, PARALLEL_CALL_STATE, Step, StepFunction } from "./types";
+import { internalHeader, workflowIdHeader } from "./types";
+import type { Client } from "../client";
+
+export class Workflow {
+  protected readonly client: Client;
+  protected readonly url: string;
+  protected stepCount = 0;
+  protected planStepCount = 0;
+  protected readonly workflowId: string;
+  protected readonly steps: Step[];
+  protected readonly nonPlanStepCount: number;
+  protected skip;
+  // to accumulate steps in Promise.all
+  protected pendingSteps: Step[] = [];
+
+  constructor({
+    client,
+    url,
+    workflowId,
+    steps,
+    skip = false,
+  }: {
+    client: Client;
+    url: string;
+    workflowId: string;
+    steps: Step[];
+    skip: boolean;
+  }) {
+    this.client = client;
+    this.url = url;
+    this.workflowId = workflowId;
+    this.steps = steps;
+    this.nonPlanStepCount = this.steps.filter((step) => !step.targetStep).length;
+    this.skip = skip;
+  }
+
+  public requestPayload() {
+    return this.steps[0].out;
+  }
+
+  public async run<TResult>(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    stepName: string,
```

Comment:
We only pass step id to Qstash but Inngest passes a hash of step name concatenated with step Id

I think we need it because at some point we will need to handle the case when a workflow endpoint is updated by the user. If there are steps with names different than what we expect, we should do something (abort or something else)

I added the step name expecting it to become relevant later